### PR TITLE
Remove python 2 support

### PR DESCRIPTION
**Pull request by basepi**
As discussed with @sethmlarson, the python client is removing support for python 2 in 8.0, so `ecs-logging-python` can follow suit. 🎉 

We'll want to move from comment-based type checking to python 3 type hints but I think that should be a separate PR.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -26,7 +25,7 @@ classifiers = [
 requires = [
   "backports.functools-lru-cache; python_version < '3.3'"
 ]
-requires-python = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*"
+requires-python = "!=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*"
```
Comment (sethmlarson) [inline pyproject.toml:28]: We can change this to `>=3.6` now.

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Looks good, only one blocking comment from me. A non-blocking comment: have you tried `pyupgrade --py36-plus` and `flynt` on the codebase? Might change some string formatting into f-strings and other modernizations.

File: pyproject.toml (hunk 1 above)
Comment (basepi) [reply pyproject.toml]: Done

**Review by sethmlarson — APPROVED** [review]

**Comment by basepi** [comment]
Ran flynt/pyupgrade and fixed up a few things

**Pull request merged**